### PR TITLE
ci(dependabot): group opentelemetry updates to a single PR

### PR DESCRIPTION
**Pull request by Edholm**
These updates generally need to be bumped together, and this should take care of that.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -10,6 +10,10 @@ updates:
     directory: /
     schedule:
       interval: daily
+    groups:
+      opentelemetry:
+        patterns:
+          - "*opentelemetry*"  # Include all dependencies in one PR
```
Comment (radhus) [inline .github/dependabot.yml:16]: Would it make sense to include `cloud.google.com/go/monitoring` as well?

**Review by radhus — APPROVED** [review]
Nice! ⭐

File: .github/dependabot.yml (hunk 1 above)
Comment (Edholm) [reply .github/dependabot.yml:16]: No it shouldn't be required. This has mostly to do with opentelemetry having so many repos that are heavily tied together :( 

I can add it if you want though?

Comment (radhus) [reply .github/dependabot.yml:16]: Ah, I thought that `cloud.google.com/go/monitoring` was linked to it somehow, but looking at its `go.mod` it looks fine! Go ahead like this!

**Review by alethenorio — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (Edholm) [reply .github/dependabot.yml:16]: Note that Dependabot won't close existing PRs when it opens a new group PR. It only does that after one or the other has been merged.

FYI

**Pull request merged**